src/main/java/se/yolean/kafka/keyvalue/http/CacheResource.java

```diff
@@ -159,24 +169,38 @@ public void write(OutputStream out) throws IOException, WebApplicationException
 
   /**
    * @return Newline separated values (no keys)
+   * @throws IOException
    */
   @GET()
   @Path("/values")
   @Produces(MediaType.TEXT_PLAIN)
-  public Response values() {
+  public Response values() throws IOException {
     requireUpToDateCache();
     Iterator<byte[]> values = cache.getValues();
 
-    StreamingOutput stream = new StreamingOutput() {
-      @Override
-      public void write(OutputStream out) throws IOException, WebApplicationException {
-        while (values.hasNext()) {
-          out.write(values.next());
-          out.write('\n');
-        }
-      }
-    };
-    return Response.ok(stream).build();
+    ByteArrayOutputStream buffer = new ByteArrayOutputStream();
+
+    while (values.hasNext()) {
+      buffer.write(values.next());
+      buffer.write('\n');
+    }
+
+    ResponseBuilder response = Response.ok(buffer);
+
+    applyOffsetHeaders(response);
+
+    return response.build();
+  }
+
+  private void applyOffsetHeaders(ResponseBuilder response) {
+    Map<TopicPartition, Long> offsets = cache.getCurrentOffsets();
+
+    offsets.forEach((key, value) -> {
+      String topic = key.topic();
+      int partition = key.partition();
+
+      response.header(UpdatesBodyPerTopic.HEADER_PREFIX + "offset-" + topic + "-" + partition, value);
```

Comment:
We probably need to slug or sanitize here sooner rather than later, to not violate any HTTP conventions. The easiest and most readable might be to just strip non-alphanumerics from the name.